skbio/alignment/_tabular_msa.py

```diff
@@ -717,7 +694,136 @@ def reindex(self, key=None, keys=None):
 
         self._keys = keys_
 
+    def _assert_valid_keys(self, keys):
+        # Hashability of keys is implicitly checked here.
+        duplicates = find_duplicates(keys)
+        if duplicates:
+            raise UniqueError(
+                "Keys must be unique. Duplicate keys: %r" % duplicates)
+
     @experimental(as_of='0.4.0-dev')
+    def append(self, sequence, minter=None, key=None):
+        """Append a sequence to the MSA.
+
+        Parameters
+        ----------
+        sequence : alphabet-aware scikit-bio sequence object
+            Sequence to be appended. Must match the dtype of the MSA and the
+            number of positions in the MSA.
+        minter : callable or metadata key, optional
+            Used to create a key for the sequence being appended. If callable,
+            it generates a key directly. Otherwise it's treated as a key into
+            the sequence metadata. If the key is a duplicate of any key
+            already in the MSA, an error is raised. Note that `minter` cannot
+            be combined with `key`.
+        key : hashable, optional
+            Key for the MSA to use for the appended sequence. Note that
+            `key` cannot be combined with `minter`.
+
+        Raises
+        ------
+        OperationError
+            If both key and minter are provided.
+        OperationError
+            If MSA has keys but no key or minter was provided for the sequence
+            being appended.
+        OperationError
+            If key was provided but the MSA does not have keys.
+        OperationError
+            If minter was provided but the MSA does not have keys.
+        TypeError
+            If the sequence object is a type that doesn't have an alphabet
+        TypeError
+            If the type of the sequence does not match the dtype of the MSA.
+        ValueError
+            If the length of the sequence does not match the number of
+            positions in the MSA.
+
+        See Also
+        --------
+        reindex
+
+        Notes
+        -----
+        The MSA is not automatically re-aligned when a sequence is appended.
+        Therefore, this operation is not necessarily meaningful on its own.
+
+        Examples
+        --------
+        >>> from skbio import DNA, TabularMSA
+        >>> msa = TabularMSA([DNA('')])
+        >>> msa.append(DNA(''))
+        >>> msa == TabularMSA([DNA(''), DNA('')])
+        True
+
+        >>> msa = TabularMSA([DNA('', metadata={'id': 'a'})], minter='id')
+        >>> msa.append(DNA('', metadata={'id': 'b'}), minter='id')
+        >>> msa == TabularMSA([DNA('', metadata={'id': 'a'}),
+        ...                    DNA('', metadata={'id': 'b'})], minter='id')
+        True
+        """
+
+        if key is not None and minter is not None:
+            raise OperationError(
```

Comment:
@jairideout How could you miss that? I mean really. I think you might be losing your edge...